android/app/src/main/kotlin/coop/polypoly/polypod/polyNav/PolyNav.kt

```diff
@@ -65,15 +69,17 @@ open class PolyNav(
                     )
             }
         }
-        contentResolver?.openInputStream(importedUrl).use { inputStream ->
-            if (inputStream == null) {
-                throw Error("File copy error")
+        coroutineScope.launch {
```

Comment:
So, I'm unsure about this being the right fix :D

This is a `suspend` function, meaning it runs asynchronously. The corresponding `await` on the feature side will return as soon as this function returns. I do think the right time to return from all that would be _once the import is finished_.

The black screens, on the other hand, were presumably caused because the suspend function was _blocking_ instead of _suspending_. IMHO, we should do the latter!

Reply:
Ah, done. This is much better, thanks!